pkg/lint/rules/values_test.go

```diff
@@ -104,7 +104,7 @@ func TestValidateValuesFileSchemaFailure(t *testing.T) {
 }
 
 func TestValidateValuesFileSchemaOverrides(t *testing.T) {
-	yaml := "username: admin"
+	yaml := "username: admin\npassword:"
```

Comment:
can you explain the reason behind this change to the existing unit tests? Could this potentially introduce breaking behaviour?

If at all possible can you please leave the existing test suite as it currently works? If this is intentional, can you please clarify why the values.yaml needs to contain a null value?

Reply:
The change to unit test TestValidateValuesFileSchemaOverrides was intentional.  TestValidateValuesFileSchemaOverrides was introduced and last modified by Issue #7984 which had the following goal:

“When support for helm lint --values was added, this support did not extend to the values file tests. This PR extends the support to values files, which lets the user schema-test the combination of the chart and the passed-in values.”

While TestValidateValuesFileSchemaOverrides does test the coalescing of a user-defined values file and the chart provided values.yaml, the unit test does not test overrides as its name specifies.  The field “password” given in the user-defined values file does not override a value in the chart provided values.yaml because the field “password” does not exist in the chart provided values.yaml.  Including the field “password” in the chart provided values.yaml verifies that an override of a field’s value has occurred with the user-provided value taking precedent.

Once the unit test was updated to take in an override, as described above, the overridden field “password” needed to be verified to ensure the correct value was used.  The function “validateValuesFile” coalesces the user-provided values file and the chart provided values.yaml, but does not return this coalesced object.  To ensure that the coalesced object has the correct value for “password” without being able to inspect the coalesced object, the field “password” in the chart provided values.yaml was given an invalid value of “null” while the user-provided values file was given a valid value of “swordfish”.  This ensured that if the function “validateValuesFile” coalesced correctly the unit test TestValidateValuesFileSchemaOverrides would pass and if the order of coalescing was incorrect that a failure would occur validating against the schema.

Reply:
The requested reasoning for the unit test change was provided above.  Can you provide feedback @bacongobbler  @jdolitsky as to how this pull request should proceed?

Reply:
@mattfarina are you able to provide feedback on this pull request?

Reply:
When we introduce new unit tests, we do not change the existing ones currently in place. Instead we add new tests to verify the new behaviour. That way we can ensure code that relied on prior behaviour still continues to work as expected. Changes like this one have introduced regressions in the past hence the concerns raised earlier.

Can you please keep the existing unit test in place and add a new unit test to verify the changes made above? A good practice to follow is described in https://golang.org/pkg/testing/, and there are examples in several areas of the helm code base, such as:

https://github.com/helm/helm/blob/6297c021cbda1483d8c08a8ec6f4a99e38be7302/pkg/action/lint_test.go#L79-L89

Reply:
Thank you for the feedback. The unit test in question was reverted and a new parameterized test was created as suggested.